spec/services/atmosphere/billing_service_spec.rb

```diff
@@ -83,22 +101,26 @@
 
       appl1 = create(:appliance, fund: cs_fund, appliance_set: wf, appliance_type: shareable_appl_type, appliance_configuration_instance: config_inst, virtual_machines: [shareable_vm])
 
+      vmf = appl1.virtual_machines.first.virtual_machine_flavor
+      vmf.set_hourly_cost_for(Atmosphere::OSFamily.first, 10)
+      vmf.reload
+
       Atmosphere::BillingService.bill_appliance(appl1, Time.now.utc, "mock billing", true)
       appl1.reload
       expect(Atmosphere::BillingLog.all.count).to eq 1
 
       cs_fund.reload
-      expect(cs_fund.balance).to eq original_balance - ((appl1.virtual_machines.first.virtual_machine_flavor.hourly_cost))
+      expect(cs_fund.balance).to eq original_balance - ((appl1.virtual_machines.first.virtual_machine_flavor.virtual_machine_flavor_os_families.first.hourly_cost))
 
       appl2 = create(:appliance, fund: cs_fund, appliance_set: wf, appliance_type: shareable_appl_type, appliance_configuration_instance: config_inst, virtual_machines: [shareable_vm])
 
       Atmosphere::BillingService.bill_appliance(appl2, Time.now.utc, "mock billing", true)
       appl2.reload
-      expect(appl1.amount_billed).to eq appl1.virtual_machines.first.virtual_machine_flavor.hourly_cost
-      expect(appl2.amount_billed).to eq (appl2.virtual_machines.first.virtual_machine_flavor.hourly_cost*0.5).round
+      expect(appl1.amount_billed).to eq appl1.virtual_machines.first.virtual_machine_flavor.virtual_machine_flavor_os_families.first.hourly_cost
+      expect(appl2.amount_billed).to eq (appl2.virtual_machines.first.virtual_machine_flavor.virtual_machine_flavor_os_families.first.hourly_cost*0.5).round
 
       cs_fund.reload
-      expect(cs_fund.balance).to eq original_balance - (appl2.virtual_machines.first.virtual_machine_flavor.hourly_cost*1.5).round
+      expect(cs_fund.balance).to eq original_balance - (appl2.virtual_machines.first.virtual_machine_flavor.virtual_machine_flavor_os_families.first.hourly_cost*1.5).round
```

Comment:
We had a discussion regarding that once - the argument (which I concurred with) is that the `expect` condition should not duplicate values which are already set in the flavor factory - this introduces needless redundancy (updating the factory would also require updating the rspec accordingly). I'm hesitant to introduce this change although if you feel strongly about it then I will do so.